### PR TITLE
feat(php): add charge server parity and interop

### PR DESCRIPTION
**Pull request by EfeDurmaz16**
## What
- add PHP charge intent models and Payment header helpers
- add PHP server-side charge verification for request binding, splits, network mismatch, and cross-route replay
- add a PHP server adapter to the TypeScript interop harness with structured bridge error codes
- add a root PHP README with local payment-check instructions
- add PHP CI coverage reporting via pcov

## Why
M1 includes PHP server-side support for MPP charge. This keeps PHP focused on server verification while matching the Rust/TypeScript charge vectors and making it selectable in the interop harness.

## Verified
- `composer validate --strict`
- `composer test` (25 tests, 59 assertions)
- `composer run test:coverage` executes tests locally but cannot produce coverage without a local coverage driver; CI installs pcov for coverage
- focused PHP interop was verified on the PHP adapter branch before cleaning the PR scope

### CODE REVIEW

**Comment by EfeDurmaz16** [comment]
@lgalabru Greptile feedback has been addressed and this is ready for maintainer review from my side.

Latest head: d00f803. The follow-ups use structured bridge errors and tighten wire consistency: leading-zero amounts are rejected, empty receipt fields are omitted consistently, the redundant PHPUnit run was removed, and the network-mismatch message now reflects the Surfpool signal. CI is green.

**Comment by EfeDurmaz16** [comment]
@lgalabru Greptile feedback has been addressed and this is ready for maintainer review from my side.

Latest head: adcf7b4. The follow-ups enforce PHP charge coverage in CI, add Base64/header/server edge-case tests, install PHP dependencies for the TypeScript interop smoke, and normalize receipt timestamps to UTC before emitting the `Z` suffix.

Verified locally with `composer test`, PHP syntax checks, and the focused TypeScript client -> PHP server `charge-basic` interop smoke. CI is green.

**Comment by EfeDurmaz16** [comment]
@lgalabru Greptile feedback has been addressed and this is ready for maintainer review from my side.

Latest head: 2fa91e0. The final pass fixes the PHP interop fixture so transaction-verifier rejections are classified as 402 Payment Required instead of falling through as 500s, and makes large u64 values fail closed instead of risking PHP integer overflow.

Verified locally with PHP tests, lint, 91.35% coverage, composer audit, TypeScript interop typecheck/vitest, TypeScript client -> PHP server interop, and Rust client -> PHP server interop.

CI is green and Greptile is back to 5/5.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -211,6 +211,33 @@ jobs:
           name: go-coverage
           path: go/coverage.out
 
+  test-php:
```
Comment (lgalabru) [inline .github/workflows/ci.yml]: Could we have introduce a `php.yml` that isolates all the php related stuffs?
We'll do the same for the other languages.

Comment (EfeDurmaz16) [reply .github/workflows/ci.yml]: I'm on it!

File: php/examples/charge-server.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    realm: $credential->challenge->realm,
+    method: $credential->challenge->method,
+    intent: $credential->challenge->intent,
+    request: $credential->challenge->request,
+    expires: $credential->challenge->expires,
+    digest: $credential->challenge->digest,
+    opaque: $credential->challenge->opaque,
+);
+
+http_response_code(200);
```
Comment (lgalabru) [inline php/examples/charge-server.php:91]: This example should simply be spinning up a web server with one gated endpoint.

the README.md explains how to spin-up this server, and send a request using `pay` cli.

**Comment by EfeDurmaz16** [comment]
addressed. php now has its own workflow, and the example is a single /paid endpoint documented with curl -> 402 and pay curl -> success.

verified locally: lint, unit tests, coverage 91.47%, ts -> php interop, and rust -> php interop.

File: php/src/Core/Base64Url.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        return $value . str_repeat('=', 4 - $remainder);
+    }
+
+    /**
+     * @param mixed $value
+     * @return mixed
+     */
+    private static function canonicalizeJson(mixed $value): mixed
```
Comment (lgalabru) [inline php/src/Core/Base64Url.php]: Is there a better way to type things in php, and avoid this `mixed` stuffs?

Comment (EfeDurmaz16) [reply php/src/Core/Base64Url.php]: will patch it shortly.

File: php/README.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+# payment required
+curl http://localhost:4567/paid
+
+# payment successful
+pay curl http://localhost:4567/paid
+```
+
+## Client compatibility matrix
+
+| Cell | Status |
```
Comment (lgalabru) [inline php/README.md]:
```suggestion
| Intent | Status |
```

File: php/README.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+| `x402/upto` | — |
+| `x402/batch-settlement` | — |
+| `mpp/charge/pull` | — |
+| `mpp/charge/push` | — |
+| `mpp/session` | — |
+| `mpp/subscription` | — |
+
+## Server compatibility matrix
+
+| Cell | Status |
```
Comment (lgalabru) [inline php/README.md]:
```suggestion
| Intent | Status |
```

File: php/README.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+```
+
+## Quick start — client (auto-402)
+
+PHP is server-side only for the current MPP roadmap. Use a client SDK or the
+`pay` CLI to complete the 402 challenge/credential flow.
+
+```bash
+brew install pay
+
```
Comment (lgalabru) [inline php/README.md:52]: can we include the commands to start the server we created in `./example` ?

**Comment by EfeDurmaz16** [comment]
@lgalabru latest PHP pass is addressed and ready for review from my side.

Latest head: 264f99e. I added the example server start commands to the README, fixed receipt challenge binding, separated pre-broadcast verification from settled receipt creation, simplified bridge error routing, and made verifier exceptions fail closed as payment verification failures.

Verified locally with PHP lint/phpstan, PHP coverage 91.14%, interop typecheck/vitest, TypeScript -> PHP interop, and Rust -> PHP interop. CI is green and Greptile is 5/5.

**Review by lgalabru — APPROVED** [review]
LGTM, thanks @EfeDurmaz16!
